langchain/memory/chat_message_histories/dynamodb.py

```diff
@@ -15,20 +17,20 @@
 logger = logging.getLogger(__name__)
 
 
-class DynamoDBMessageDB(MessageDB):
-    def __init__(self, table_name: str):
+class DynamoDBChatMessageHistory(ChatMessageHistoryBase):
```

Comment:
Not sure if the specific message history classes should be included here. Users might have a different db structure, so not sure if this belongs here. Ok, to utilize the changes, and keep this in your project. Also, ok to include an example in the unit tests or documentation, so users can follow.

Reply:
There are too many storage options, for example different databases, filesystems, cloud storage etc. These classes are not core to this project, so not sure if they belong here. Adding examples to the documentation should suffice for now, if there is a need for these in future, we can always add them in a separate PR, leaving them out is not a one-way door.

Reply:
I agree that there are too many storage options, but I don't agree that that means that we should not include any here.

Based on that same argument of having too many options what is the reason to have any functionality in this library at all? We could just say that prompt engineering has too many options and just write a documentation on how one can write applications with prompt engineering instead of writing `langchain`.

It makes sense to write things into the library that is expected to be used by many, so they don't all need to re-implement the wheel.

But yes, for those storage options which are very niche, the argument not to include them could be valid.

Redis is definitely not niche, so that should be included but DynamoDB could be considered as niche and not included.

Ps.: I would be curious as to how many users of this library will be able to write their own driver for their own storage option. I would guess that probably less than 30%. :-)

Either way this decision (what to include in the library and what not to) probably belongs to @hwchase17.